src/middleware/jwt.ts

```diff
@@ -1,19 +1,73 @@
 import Jwt from 'express-jwt'
-import config from '../config'
 import { Request } from 'express'
+import { JwtPayload, Secret, sign, SignOptions, decode } from 'jsonwebtoken'
+import config from '../config'
 
-export default Jwt({
-  secret: config.get('jwt.public'),
-  algorithms: [config.get('jwt.algorithm')],
-  credentialsRequired: true,
-  getToken: function fromHeaderOrQuerystring (req: Request) {
-    if (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer') {
-      return req.headers.authorization.split(' ')[1]
-    } else if (req.query && req.query.token) {
-      return req.query.token
-    } else if (req.cookies && req.cookies.access_token) {
-      return req.cookies.access_token
-    }
-    return null
+interface VerifyToken {
+  secretOrPublic: string
+  algorithms: string
+  isRequired?: boolean
+}
+
+interface CreateToken extends SignOptions {
+  data: object
+  secret: Secret
+}
+
+const getToken = (req: Request) => {
+  if (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer') {
```

Comment:
I think this can be shortened to
```ts
if (req.headers?.authorization?.split(' ')?.[0] === 'Bearer')
```
but it's up to you since not everyone is used to read that null safety syntax

Reply:
thanks i agree